.drone.yml

```diff
@@ -1,297 +1,391 @@
-    last_successful: true
-    token: ${DRONE_TOKEN}
-    secrets:
-      - drone_token
+    token:
+      from_secret: DRONE_TOKEN
     params:
       - info-service-data-version.env
     repositories:
-      - UKHomeOffice/hocs-data@master
+      - UKHomeOffice/hocs-data@main
+    depends_on:
+      - generate & tag build
+      - env-file
     when:
-      event: deployment
-      environment: [hocs-qax]
+      event:
+        - promote
+      target:
+        - release
 
-  trigger-wcs-downstream-data-build:
+  - name: WCS downstream build
     image: plugins/downstream
     server: https://drone-gh.acp.homeoffice.gov.uk
-    fork: true
-    token: ${DRONE_TOKEN}
+    fork: false
     deploy: wcs-qa
     last_successful: true
-    secrets:
-      - drone_token
+    token:
+      from_secret: DRONE_TOKEN
     params:
       - info-service-data-version.env
     repositories:
-      - UKHomeOffice/hocs-data-wcs@master
-    when:
-      event: deployment
-      environment: [wcs-qa]
-
-  clone-kube-project:
-    image: plugins/git
-    commands:
-      - git clone https://github.com/UKHomeOffice/kube-hocs-info-service.git
+      - UKHomeOffice/hocs-data@main
+    depends_on:
+      - generate & tag build
+      - env-file
     when:
-      event: [push, tag, deployment]
+      event:
+        - promote
+      target:
+        - release
 
-  deploy-to-cs-dev-from-build-number:
-    group: deploy-to-dev
+  - name: deploy to cs-qa
     image: quay.io/ukhomeofficedigital/kd:v1.16.0
     environment:
-      - ENVIRONMENT=cs-dev
-      - VERSION=build-${DRONE_BUILD_NUMBER}
-      - HOCS_INFO_SERVICE_DATA_VERSION=latest
+      - ENVIRONMENT=cs-qa
       - KUBE_SERVER=https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
     secrets:
-      - hocs_info_service_dev_cs
+      - hocs_info_service_cs_qa
     commands:
+      - source version.txt
+      - echo $VERSION
       - cd kube-hocs-info-service
       - ./deploy.sh
+    depends_on:
+      - clone kube repo
+      - CS downstream build
     when:
-      branch: master
-      event: [push, tag]
+      event:
+        - promote
+      target:
+        - release
 
-  deploy-to-wcs-dev-from-build-number:
-    group: deploy-to-dev
+  - name: deploy to wcs-qa
     image: quay.io/ukhomeofficedigital/kd:v1.16.0
     environment:
-      - ENVIRONMENT=wcs-dev
-      - VERSION=build-${DRONE_BUILD_NUMBER}
-      - HOCS_INFO_SERVICE_DATA_VERSION=latest
+      - ENVIRONMENT=wcs-qa
       - KUBE_SERVER=https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
     secrets:
-      - hocs_info_service_dev_wcs
+      - hocs_info_service_cs_qa
     commands:
+      - source version.txt
+      - echo $VERSION
       - cd kube-hocs-info-service
       - ./deploy.sh
+    depends_on:
+      - clone kube repo
+      - WCS downstream build
     when:
-      branch: master
-      event: [push, tag]
+      event:
+        - promote
+      target:
+        - release
+
 
-  deployment:
+  - name: deploy to cs-dev
```

Comment:
Can this be reordered - as our environment is dev -> qa -> demo it would be good to have this deploy steps in that order too.

Reply:
Completed 👍🏼